src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1715,35 +1715,52 @@ abstract class RefChecks extends Transform {
+     *  except that the rule is relaxed when the method has exactly one boolean parameter
      *  and it is the first parameter, such as `assert(false, msg)`.
      */
     private def checkNamedBooleanArgs(fn: Tree, args: List[Tree]): Unit = {
       val sym = fn.symbol
       def applyDepth: Int = {
         def loop(t: Tree, d: Int): Int =
           t match {
-            case Apply(f, _) => loop(f, d+1)
+            case Apply(t, _) => loop(t, d+1)
             case _ => d
           }
         loop(fn, 0)
       }
-      def isAssertParadigm(params: List[Symbol]): Boolean = !sym.isConstructor && !sym.isCaseApplyOrUnapply && {
-        params match {
-          case h :: t => h.tpe == BooleanTpe && !t.exists(_.tpe == BooleanTpe)
-          case _ => false
-        }
-      }
-      if (settings.lintNamedBooleans && !sym.isJavaDefined && !args.isEmpty) {
+      if (settings.warnNamedLiteral.value && !sym.isJavaDefined && !args.isEmpty) {
+        def isUnnamedArg(t: Tree) = t.hasAttachment[UnnamedArg.type]
         val params = sym.paramLists(applyDepth)
-        if (!isAssertParadigm(params))
-          foreach2(args, params)((arg, param) => arg match {
-            case Literal(Constant(_: Boolean))
-            if arg.hasAttachment[UnnamedArg.type] && param.tpe.typeSymbol == BooleanClass && !param.deprecatedParamName.contains(nme.NO_NAME) =>
-              runReporting.warning(arg.pos, s"Boolean literals should be passed using named argument syntax for parameter ${param.name}.", WarningCategory.LintNamedBooleans, sym)
-            case _ =>
-          })
+        val numBools = params.count(_.tpe == BooleanTpe)
+        def onlyLeadingBool = numBools == 1 && params.head.tpe == BooleanTpe
+        val checkable = if (settings.Ylint.value) numBools > 0 && !onlyLeadingBool else numBools >= 2
+        if (checkable) {
+          val (unnamed, numSuspicious) = args.lazyZip(params).iterator
+            .foldLeft((List.empty[(Tree, Symbol)], 0)) { (acc, ap) =>
+              ap match {
+                case (arg, param) if param.tpe.typeSymbol == BooleanClass && !param.deprecatedParamName.contains(nme.NO_NAME) =>
+                  arg match {
+                    case Literal(Constant(_: Boolean)) =>
+                      if (isUnnamedArg(arg)) (ap :: acc._1, acc._2 + 1) else acc
+                    case _ =>
+                      if (arg.symbol != null && arg.symbol.isDefaultGetter) (acc._1, acc._2 + 1) else acc
+                  }
+                case _ => acc
+              }
+            }
+          val warn = !unnamed.isEmpty && (settings.Ylint.value || numSuspicious >= 2)
+          if (warn)
+            unnamed.reverse.foreach {
+              case (arg, param) =>
+                val msg = s"Boolean literals should be passed using named argument syntax for parameter ${param.name}."
+                val action = runReporting.codeAction("name boolean literal", arg.pos.withEnd(arg.pos.start), s"${param.name} = ", msg)
```

Comment:
`arg.pos.focusStart`

Reply:
That is not a range position, but I suppose it doesn't matter? or could it matter? Would have to refresh memory on position invariants, transparency, how edits are applied, etc.